upload/common/upload_area.py

```diff
@@ -197,21 +197,25 @@ def store_file(self, filename, content, content_type):
         # Compute client-side checksums for the file being uploaded
         checksum_handler = ClientSideChecksumHandler(data=content)
         clientside_checksums = checksum_handler.get_checksum_metadata_tag()
-        file = UploadedFile.create(upload_area=self, checksums=clientside_checksums, name=filename,
-                                   content_type=str(media_type), data=content)
-        checksum_id = str(uuid.uuid4())
-        checksum_event = ChecksumEvent(file_id=file.db_id,
-                                       checksum_id=checksum_id,
-                                       status="CHECKSUMMING")
-        checksum_event.create_record()
-
-        checksums = DssChecksums(s3_object=file.s3object)
-        checksums.compute(report_progress=True)
-        checksums.save_as_tags_on_s3_object()
-        file.checksums = dict(checksums)
-
-        checksum_event.status = "CHECKSUMMED"
-        checksum_event.update_record()
+
+        file, file_newly_created = UploadedFile.create(upload_area=self, checksums=clientside_checksums, name=filename,
```

Comment:
A returned tuple seems a little messy here. Why not set file.file_newly_created during UploadedFile.create and access it after method invocation?

Reply:
Done